### DISCUSSION TITLE
To change/move the canvas view to focus particular node on its the selection

### DISCUSSION THREAD

**Discussion by Abhay0294** [eclipse-glsp/glsp#761] [Q&A] [answered]
Hi,

I have the following requirement:
1.I'm having a custom extension which is holding elements which will be dragged & dropped to the GLSP canvas
2.My requirement is that when I click on the element(say xyz) on the custom extension holding the elements as show in the snippet, the GLSP canvas should focus/change its bounds to focus the selected element on the canvas in center, as shown in the second snippet.

Requesting for the suggestions to implement the same. Thank you.

<img width="940" alt="Requirement_10_10_1" src="https://user-images.githubusercontent.com/57217189/194805249-62357945-08c2-4a60-9731-15ff7700b219.png">
<img width="929" alt="Requirement_10_10_2" src="https://user-images.githubusercontent.com/57217189/194805261-f4889fe0-373a-4d47-950a-96f46b88bc23.png">

**Comment by planger** [comment, marked as answer]
Hi,

this can be done by dispatching a [`CenterAction`](https://github.com/eclipse-glsp/glsp/blob/master/PROTOCOL.md#2811-centeraction) with the respective GModel element ID on click from your UI extension.

**Reply by Abhay0294** [reply]
Thanks a lot @planger.
